docs/src/main/docs/guides/91_mp-tutorial.adoc

```diff
@@ -0,0 +1,1258 @@
+
+import java.io.IOException;
+
+import static io.helidon.config.ConfigSources.classpath;
+import static io.helidon.config.ConfigSources.file;
+
+public final class Main {
+
+    private Main() { }
+
+    public static void main(final String[] args) throws IOException {
+        Server server = startServer();
+        System.out.println("http://localhost:" + server.port() + "/greet");
+    }
+
+    private static Config buildConfig() {
+        return Config.builder()
+                .sources(
+                        file("conf/mp.yaml") <1>
+                            .pollingStrategy(PollingStrategies::watch)
+                            .optional(),
+                        classpath("META-INF/microprofile-config.properties"))
+                .build();
+    }
+
+    static Server startServer() {
+        return Server.builder()
+                     .config(buildConfig()) <2>
+                     .build()
+                     .start();
+    }
+
+}
+----
+
+These updates introduce a new configuration file `conf/mp.yaml` which 
+ can be kept outside of the JAR file, and which can be used to provide
+ additional configuration data and/or to override the data in the
+ `microprofile-config.properties` which will be packaged in the JAR file.
+
+Layered configuration like this, where the configuration is read from 
+ more than one source with overrides, is commonly used in cases where 
+ you want to have different configurations for different environments
+ like development, test and production.
+
+TIP: Learn much more about configuration in <<config/06_advanced-configuration.adoc,Advanced Configurtion Topics>>.
+
+Create this file with the following content: 
+
+[source,yaml]
+.Initial mp.yaml configuration file
+----
+app:
+  greeting: "Hallo"
+----
+
+<1> The new `buildConfig()` method defines two sources for configuration information, first the
+ new `mp.yaml` file, and this is marked as optional, and has a "polling
+ strategy" of "watch".  The polling strategy tells Helidon to watch for 
+ any updates to this file and to dynamically update the configuration.
+ The second entry is the `microprofile-config.properties` file from the
+ classpath.
+<2> Notice that the `startServer()` method has been updated to use a
+ Config Builder, which is implemented in the new method `buildConfig()`.
+
+Rebuild the application and run it.  After making some requests, update
+ the greeting in the `mp.yaml` and observe that new requests pick up
+ the new value without the need for a restart of the application.
+
+TIP: To learn more about Helidon MP configuration please see the 
+ <<config/01_introduction.adoc,Config>> section of the documentation.
+
+== Extending the application
+
+In this section, the application will be extended to add a POST 
+ service which will allow users to update the greeting and a
+ second GET service which will accept a parameter.
+
+Here are the two new methods to add to `GreetResource.java`:
+
```

Comment:
This needs new imports

import javax.ws.rs.Consumes;
import javax.ws.rs.PUT;
import javax.ws.rs.PathParam;
import javax.ws.rs.core.Response;